intellij-ext/modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt

```diff
@@ -0,0 +1,65 @@
+package com.metalbear.mirrord
+
+import com.intellij.openapi.actionSystem.*
+import com.intellij.openapi.actionSystem.ex.ComboBoxAction
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.newvfs.BulkFileListener
+import com.intellij.openapi.vfs.newvfs.events.VFileEvent
+import javax.swing.JComponent
+
+class MirrordConfigDropDown : ComboBoxAction() {
+    companion object {
+        private var project: Project? = null
+        private val configPaths by lazy { arrayListOf<String>() }
```

Comment:
overall change I think that `configPaths` should be `Option` (or Kotlin alternative?) then that's our flag for initialization

Reply:
also if you do it that way it can be just an accessor instead of a function

Reply:
i think for options in kotlin we need to depend on Arrow? and i believe null values in kotlin have the same meaning as the option type